main.js

```diff
@@ -1,123 +1,169 @@
+ * @param {object} target The target on which the action is executed
+ *
+ * @returns {object} the channel details
  */
-const sendToIOTA = () => {
+async function sendToIOTA (action, target) {
   // Hash the action with SHA256
-  const sha256 = hashJs.sha256()
-    .update(jsonSortify(action))
-    .digest('hex');
+  const sha256 = hashJs.sha256().update(jsonSortify(action)).digest('hex');
   logger.debug(`Action SHA256: ${sha256}`);
 
-  // Use pre-recorded MAM state, else use a new one
-  let mamState = Mam.changeMode(Mam.init(NODE_ADDRESS), 'public');
-  if(target.customFields && target.customFields.iotaMamState) {
-    mamState = target.customFields.iotaMamState;
+  // We need to check whether a new channel is needed to be created or not
+  let channelDetails =
+    target.customFields && target.customFields.iotaAnchoringChannel;
+  let channel;
+  // The anchorage to be used to anchor this message (hash)
+  let anchorageID;
+
+  // Any Node connection params should be under the options object
+  let options;
+  if (NODE_ADDRESS) {
+    options = {
+      node: NODE_ADDRESS
+    };
   }
 
-  // Encode the payload
-  try {
-    const { root, payload, address } = Mam.create(mamState, asciiToTrytes(sha256));
-    return Mam.attach(payload, address, DEPTH, MWM).then(() => {
-      logger.debug(`root: ${root}`);
-      return { root, mamState };
-    }).catch((e) => {
-      // e is a string
-      throw new Error(e);
-    });
-  } catch (e) {
-    return Promise.reject('Failed to create and/or attach: ' + e.message + ' - ' + e.stack);
+  if (channelDetails) {
+    const channelID = channelDetails.channelID;
+    const seed = channelDetails.seed;
+    channel = await IotaAnchoringChannel.fromID(channelID, options).bind(seed);
+
+    anchorageID = channelDetails.nextAnchorageID;
+  } else {
+    channelDetails = {};
+
+    // A new channel is bound and created. Node not specified -> Chrysalis mainnet
+    channel = await IotaAnchoringChannel.bindNew(options);
+
+    channelDetails.channelID = channel.channelID;
+    // This seed will also be used later to anchor more messages to the channel
+    channelDetails.seed = channel.seed;
+    // In our case Author Pub Key === Subscriber Pub Key
+    channelDetails.publicKey = channel.subscriberPubKey;
+
+    anchorageID = channel.firstAnchorageID;
   }
-};
+
+  const anchoringResult = await channel.anchor(
+    Buffer.from(sha256),
+    anchorageID
+  );
+
+  // The anchorage that will be used for the next one
+  const nextAnchorageID = anchoringResult.msgID;
+  channelDetails.nextAnchorageID = nextAnchorageID;
+
+  return channelDetails;
+}
 
 /**
- * Update the target with the IOTA MAM root, if it does not exist, and always update the MAM state.
+ * Update the target with the channel details
  *
- * @param {object} data - root and mamState from sendToIOTA().
- * @returns {Promise} Promise that resolves once the target is updated.
+ * @param {object} target the item
```

Comment:
Add a hyphen between the param name and it's description, like `channelDetails` already has here.

Reply:
fixed